### DISCUSSION TITLE
How to create custom bundle for Homepage.

### DISCUSSION THREAD

**Discussion by dheerajKsharma** [breezefront/community#115] [1. Help] [answered]
How to create custom bundle for Homepage. Your example of Blog custom bundle doesn't help out much. Can you share approach for homepage custom bundle.

**Comment by vovayatsyuk** [comment]
Use unique bundle name for your components: https://breezefront.com/docs/components

Example:

```js
<referenceBlock name="breeze.js">
  <arguments>
    <argument name="bundles" xsi:type="array">
      <item name="my-bundle" xsi:type="array">
        <item name="items" xsi:type="array">
          <item name="Vendor_Module/js/component" xsi:type="string">Vendor_Module/js/breeze/file</item>
          ...
        </item>
      </item>
    </argument>
  </arguments>
</referenceBlock>
```

**Reply by dheerajKsharma** [reply]
Sorry, For not specifying about the area of bundle. Actually we are creating Custom CSS Bundle using https://breezefront.com/docs/custom-styles But not able to perform it successfully for homepage. 
We have create cms_index_index.xml inside Magento_Theme folder. Then created home.less file as well as deferred-home.less in the mentioned folder structure. But when we are running content deploy It gives error of not finding home.less file.

**Reply by vovayatsyuk** [reply]
Please show me your XML, folder structure of your theme, and the contents of these two less files.

**Reply by dheerajKsharma** [reply]
Here you go:

Folder Structure: 
![MicrosoftTeams-image (2)](https://user-images.githubusercontent.com/132253509/235432773-a53daf8a-55d5-4a5d-9876-8a0898e4e720.png)

File: Cms_index_index.xml: 
![MicrosoftTeams-image (3)](https://user-images.githubusercontent.com/132253509/235432786-e52521cf-959b-489e-89b3-01f382642efa.png)

File: home.less: 
![MicrosoftTeams-image (4)](https://user-images.githubusercontent.com/132253509/235432802-e05186b8-6838-4c54-89d7-0bc3bd475b10.png)


File: Deferred_home.less
`@import 'home.less';`
`@critical: false;`

Let me know If you need anything else.
Thanks

**Reply by vovayatsyuk** [reply]
You added the following line:

```
@import (reference) 'custom/abstracts/_all.less';
```

But I don't see this file on your screenshots. (We don't have this line in the docs.)

**Reply by dheerajKsharma** [reply]
**It's in root web/css folder.**

![Screenshot from 2023-05-01 14-57-16](https://user-images.githubusercontent.com/132253509/235434858-46738c92-bca9-488e-ad74-54377681b5b0.png)

**Reply by vovayatsyuk** [reply, marked as answer]
You have to put all less files in the root.

**Reply by vovayatsyuk** [reply]
Screenshot from docs:

![image](https://user-images.githubusercontent.com/306080/235435252-e2964a12-6ba7-41e8-af38-59b812c12fee.png)

**Reply by dheerajKsharma** [reply]
Thanks, It worked now. Created our first custom CSS Bundle.
Cheers :)
